cmd/init.go

```diff
@@ -109,90 +111,92 @@ var initCmd = &cli.Command{
 	Usage: "create a new gqlgen project",
 	Flags: []cli.Flag{
 		&cli.BoolFlag{Name: "verbose, v", Usage: "show logs"},
-		&cli.StringFlag{Name: "config, c", Usage: "the config filename"},
+		&cli.StringFlag{Name: "config, c", Usage: "the config filename", Value: "gqlgen.yml"},
 		&cli.StringFlag{Name: "server", Usage: "where to write the server stub to", Value: "server.go"},
 		&cli.StringFlag{Name: "schema", Usage: "where to write the schema stub to", Value: "graph/schema.graphqls"},
 	},
 	Action: func(ctx *cli.Context) error {
 		configFilename := ctx.String("config")
 		serverFilename := ctx.String("server")
+		schemaFilename := ctx.String("schema")
 
 		pkgName := code.ImportPathForDir(".")
 		if pkgName == "" {
 			return fmt.Errorf("unable to determine import path for current directory, you probably need to run go mod init first")
 		}
 
-		if err := initSchema(ctx.String("schema")); err != nil {
-			return err
+		// check schema and config don't already exist
+		if fileExists(configFilename) {
+			return fmt.Errorf("%s already exists\n", configFilename)
+		}
+		if fileExists(schemaFilename) {
+			return fmt.Errorf("%s already exists\n", schemaFilename)
 		}
-		if !configExists(configFilename) {
-			if err := initConfig(configFilename, pkgName); err != nil {
-				return err
-			}
+		if fileExists(serverFilename) {
+			return fmt.Errorf("%s already exists\n", serverFilename)
+		}
+		_, err := config.LoadConfigFromDefaultLocations()
+		if err == nil {
+			return fmt.Errorf("gqlgen.yml already exists in a parent directory\n")
 		}
 
-		GenerateGraphServer(serverFilename)
-		return nil
-	},
-}
+		// create config
+		fmt.Println("Creating", configFilename)
+		if err := initFile(configFilename, executeConfigTemplate(pkgName)); err != nil {
+			return err
+		}
 
-func GenerateGraphServer(serverFilename string) {
-	cfg, err := config.LoadConfigFromDefaultLocations()
-	if err != nil {
-		fmt.Fprintln(os.Stderr, err.Error())
-	}
+		// create schema
+		fmt.Println("Creating", schemaFilename)
+		if err := initFile(schemaFilename, schemaDefault); err != nil {
+			return err
+		}
 
-	if err := api.Generate(cfg, api.AddPlugin(servergen.New(serverFilename))); err != nil {
-		fmt.Fprintln(os.Stderr, err.Error())
-	}
+		// create the package directory with a temporary file so that go recognises it as a package
+		// and autobinding doesn't error out
+		tmpPackageNameFile := "graph/model/_tmp_gqlgen_init.go"
+		if err := initFile(tmpPackageNameFile, "package model"); err != nil {
+			return err
+		}
+		defer os.Remove(tmpPackageNameFile)
 
-	fmt.Fprintf(os.Stdout, "Exec \"go run ./%s\" to start GraphQL server\n", serverFilename)
-}
+		var cfg *config.Config
+		if cfg, err = config.LoadConfig(configFilename); err != nil {
+			panic(err)
```

Comment:
Do we need the stack trace here? If not, I think `log.Fatal` is preferred.

Reply:
I assume the panic is necessary to handle the deferred cleanup of the above files.

Reply:
Might be cleaner (also for the end-user) to call `os.Remove` manually before exiting.

Reply:
it's a panic because this should never ever happen - it's our own template we're parsing